### PR TITLE
refactor: passing native_lib_path to mlperf_backend_matches_hardware()

### PR DESCRIPTION
**Pull request by mohitmundhragithub**
Adding the API Changes for all the backends.
Need this change for QTI backend, specifically. However, the api needs to be modified for all the backends.

### CODE REVIEW

File: flutter/cpp/backends/external.h
Comment (anhappdev) [inline flutter/cpp/backends/external.h:101]: I would place `native_lib_path` right after `lib_path` since they are similar.

File: flutter/cpp/c/backend_c.h
Comment (anhappdev) [inline flutter/cpp/c/backend_c.h:39]: Does it make sense to add `native_lib_path` as a new field to `mlperf_device_info_t` struct? This way we will not need to update the function `mlperf_backend_matches_hardware()` in other backends.

File: flutter/cpp/flutter/dart_backend_match.h
Comment (anhappdev) [inline flutter/cpp/flutter/dart_backend_match.h:15]: I would place `native_lib_path` right after `lib_path` since they are similar.

**Review by anhappdev — COMMENTED** [review]
I renamed the title to make it a bit more clear what this PR is about.

**Review by anhappdev — DISMISSED** [review]
LGTM

**Comment by mohitmundhragithub** [comment]
can we merged this now?

**Comment by anhappdev** [comment]
Currently, the iOS test failed because of an open issue related to Homebrew (see https://github.com/Homebrew/homebrew-core/issues/135315)

Please cherry-pick this commit https://github.com/mlcommons/mobile_app_open/pull/746/commits/80b99c59bee9afd793c0c5be4ce912c26a8de37f into this PR else it will fail the CI test when merging.

**Comment by anhappdev** [comment]
@mohitmundhragithub Fell free to approve and merge.

**Comment by mohitmundhragithub** [comment]
@freedomtan @anhappdev, can one of you please approve? Anh and my Approvals won't work because both of us have some commits.

**Comment by anhappdev** [comment]
@mohitmundhragithub If you are currently blocked by this PR and want to merge quickly. You can merge the master branch into this. I will re-approve and it should work. The approver must not be the last pusher.

**Review by anhappdev — APPROVED** [review]

**Comment by mohitmundhragithub** [comment]
> @mohitmundhragithub If you are currently blocked by this PR and want to merge quickly. You can merge the master branch into this. I will re-approve and it should work. The approver must not be the last pusher.

merged.

**Comment by anhappdev** [comment]
Hmm. It didn't work :(. Maybe a merge is not counted as a push because GitHub still says I am the last pusher.

**Comment by anhappdev** [comment]
@freedomtan Please help approve this PR. Thank you.

**Review by freedomtan — APPROVED** [review]

**Pull request merged**
